cmd/lintwalk/main.go

```diff
@@ -2,18 +2,43 @@ package lintwalk
 
 import (
 	"flag"
+	"go/build"
+	"go/parser"
+	"go/token"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"regexp"
+	"runtime"
+	"strings"
 )
 
+func packagePath() []string {
+	res := []string{runtime.GOROOT(), build.Default.GOPATH}
+
+	if p, ok := os.LookupEnv("GOPATH"); ok {
+		res = append(res, strings.Split(p, ":")...)
+	}
```

Comment:
I like the intention of handling more corner cases, but ~ no one uses multiple GOPATHs since most Go tools and linters can't handle it. Text editors can't do it either (not even sure IDEs can).
This feature can even be deprecated in near future (especially since vgo is approaching).

Also note that about 70% of Go programmers do not know about `:` and GOPATH lists, so you should comment what you're doing in that loop and why it's required.
The other option is to remove it and ignore it's existence.